packages/fast-usdc/test/cli/transfer.test.ts

```diff
@@ -0,0 +1,169 @@
+  const out = mockOut();
+  const file = mockFile(path);
+
+  // @ts-expect-error mocking partial Console
+  await transfer.transfer(file, '1500000', 'noble1234', out);
+
+  t.is(
+    out.getErrOut(),
+    `No config found at ${path}. Use "config init" to create one, or "--home" to specify config location.\n`,
+  );
+  t.is(out.getLogOut(), '');
+});
+
+test('Transfer registers the noble forwarding account if it does not exist', async t => {
+  const path = 'config/dir/.fast-usdc/config.json';
+  const nobleApi = 'http://api.noble.test';
+  const nobleToAgoricChannel = 'channel-test-7';
+  const config = {
+    agoricRpc: 'http://rpc.agoric.test',
+    nobleApi,
+    nobleToAgoricChannel,
+    nobleSeed: 'test noble seed',
+    ethRpc: 'http://rpc.eth.test',
+  };
+  const out = mockOut();
+  const file = mockFile(path, JSON.stringify(config));
+  const agoricSettlementAccount = 'agoric123456';
+  const settlementAccountVstoragePath = 'published.fastUSDC.settlementAccount';
+  const vstorageMock = makeVstorageMock({
+    [settlementAccountVstoragePath]: agoricSettlementAccount,
+  });
+  const amount = '150';
+  const destination = 'dydx1234';
+  const nobleFwdAccountQuery = `${nobleApi}/noble/forwarding/v1/address/${nobleToAgoricChannel}/${agoricSettlementAccount}+${destination}`;
+  const fetchMock = makeFetchMock({
+    [nobleFwdAccountQuery]: { address: 'noble123456', exists: false },
+  });
+  const nobleSignerAddress = 'noble09876';
+  const signerMock = makeMockSigner();
+
+  await transfer.transfer(
+    file,
+    amount,
+    destination,
+    // @ts-expect-error mocking console
+    out,
+    fetchMock.fetch,
+    vstorageMock.vstorage,
+    { signer: signerMock.signer, address: nobleSignerAddress },
+  );
+
+  t.is(vstorageMock.getQueryCounts()[settlementAccountVstoragePath], 1);
+  t.is(fetchMock.getQueryCounts()[nobleFwdAccountQuery], 1);
+  t.deepEqual(signerMock.getSigned(), [
+    nobleSignerAddress,
+    [
+      {
+        typeUrl: '/noble.forwarding.v1.MsgRegisterAccount',
+        value: {
+          channel: nobleToAgoricChannel,
+          recipient: `${agoricSettlementAccount}+${destination}`,
+          signer: nobleSignerAddress,
+        },
+      },
+    ],
+    {
+      amount: [
+        {
+          amount: '20000',
+          denom: 'uusdc',
+        },
+      ],
+      gas: '200000',
+    },
+    'Register Account Transaction',
+  ]);
+});
+
+const makeMockEthProvider = () => {
+  const txnArgs: any[] = [];
+  const provider = {
+    getTransactionCount: () => {},
+    estimateGas: () => {},
+    getNetwork: () => ({ chainId: 123 }),
+    getFeeData: () => ({ gasPrice: 1 }),
+    broadcastTransaction: (...args) => {
+      txnArgs.push(args);
+      return {
+        blockNumber: 9000,
+      };
+    },
+    getTransactionReceipt: () => ({
+      blockNumber: 9000,
+      hash: 'SUCCESSHASH',
+      confirmations: () => [9000],
+      logs: [],
+    }),
+  };
+
+  return { provider, getTxnArgs: () => txnArgs };
```

Comment:
That allows the caller to modify `txnArgs`.

```suggestion
  return { provider, getTxnArgs: () => harden([...txnArgs]) };
```

provider should be hardened before it's returned too. Perhaps not critical in test code, but it's only a few characters and it's a good habit.

Reply:
Done